### PR TITLE
added changes to enable building image on s390x

### PR DESCRIPTION
**Pull request by psshehzin**
Added steps to use mod_auth_openidc and dependent rpms for docker image build in s390x.

### CODE REVIEW

**Comment by Fryguy** [comment]
offhand, this looks good.  Minor, but the indenting needs to be cleaned up for readability...thinking something like:

```
RUN if [ $(uname -m) != "s390x" ] ; then \
      dnf -y --disableplugin=subscription-manager --setopt=tsflags=nodocs install \
        http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-stream-repos-8-2.el8.noarch.rpm \
        http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm && \
      dnf config-manager --setopt=appstream*.exclude=*httpd* --save && \
      dnf -y --disableplugin=subscription-manager module enable mod_auth_openidc && \
      dnf -y --disableplugin=subscription-manager install mod_auth_openidc; \
    else \
      dnf -y install \
        /opt/app-root/src/bin-rpm-dir/cjose-0.6.1-2.el8.s390x.rpm \
        /opt/app-root/src/bin-rpm-dir/cjose-devel-0.6.1-2.el8.s390x.rpm \
        /opt/app-root/src/bin-rpm-dir/mod_auth_openidc-2.3.7-11.el8.s390x.rpm && \
      rm -rf /opt/app-root/src/bin-rpm-dir; \
    fi && \
    dnf -y --disableplugin=subscription-manager --setopt=tsflags=nodocs install \
...
```

**Comment by Fryguy** [comment]
@bdunne Please review.

File: Dockerfile [hunk 1]
```diff
@@ -16,14 +16,22 @@ LABEL name="Httpd" \
...
+        http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm && \
+      dnf config-manager --setopt=appstream*.exclude=*httpd* --save && \
+      dnf -y --disableplugin=subscription-manager module enable mod_auth_openidc && \
+      dnf -y --disableplugin=subscription-manager install mod_auth_openidc; \
+    else \
+      dnf -y install \
+        /opt/app-root/src/bin-rpm-dir/cjose-*.s390x.rpm \
+        /opt/app-root/src/bin-rpm-dir/cjose-devel-*.s390x.rpm \
+        /opt/app-root/src/bin-rpm-dir/mod_auth_openidc-*.s390x.rpm && \
+      rm -rf /opt/app-root/src/bin-rpm-dir; \
```
Comment (bdunne) [inline Dockerfile]: I wonder if we can leverage build secrets here to mount the RPMs at build time.

Comment (bdunne) [reply Dockerfile]: Unfortunately the `rm -rf` won't actually save any size from the images if these will be copied in another build step.

Comment (Fryguy) [reply Dockerfile]: Oh good point...I was thinking these would be a volume, but I forgot you can't do volumes at build time :(

**Comment by Fryguy** [comment]
@psshehzin How do you expect to run this?  You can't attach volumes during build, so how are you getting the /opt/app-root/src/bin-rpm-dir ?

File: Dockerfile (hunk 1 above)
Comment (SriramaSharma) [reply Dockerfile]: Other feasible option is to push these s390x built rpms into a repo and in the above step just directly do a `wget and install` or `add the repo and do yum/dnf install`. In either case, the decision to be taken is - where do we push/host those rpms ?

**Comment by psshehzin** [comment]
@Fryguy  i did this by introducing the copy step dynamically at the time of s390x build as you had suggested so that the x86 builds would be unaffected. Also the three rpms combined take around 500kb . 
` sed -i "/^RUN if [ $(uname -m) != \"s390x\" ]*/i COPY --from=$rpm_source_image /workspace/bin-rpm-dir /opt/app-root/src/bin-rpm-dir\n" Dockerfile `

**Comment by Fryguy** [comment]
Ah ok, that makes sense now.  Even so, I think @bdunne is right in that the `rm-rf` step will mostly be useless, but I can't see any way around that for now, and I'd rather move forward, so I'm good with that part.

File: Dockerfile (hunk 1 above)
Comment (Fryguy) [reply Dockerfile]: Right, we were also discussing having a proper RPM repo somewhere.  The way it is in this PR, though, is probably the simplest way forward until we get there.

**Comment by SriramaSharma** [comment]
@Fryguy I agree on the comment from @bdunne. But until we figure out a more comprehensive way like having a separate repository defined, can we go ahead with this if @psshehzin has addressed all other changes and if it looks good ? 

> Ah ok, that makes sense now. Even so, I think @bdunne is right in that the `rm-rf` step will mostly be useless, but I can't see any way around that for now, and I'd rather move forward, so I'm good with that part.

**Review by Fryguy — APPROVED** [review]
LGTM - also want @bdunne to review.

File: Dockerfile [hunk 2]
```diff
@@ -16,14 +16,22 @@ LABEL name="Httpd" \
...
+RUN if [ $(uname -m) != "s390x" ] ; then \
+      dnf -y --disableplugin=subscription-manager --setopt=tsflags=nodocs install \
+        http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-stream-repos-8-2.el8.noarch.rpm \
+        http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm && \
+      dnf config-manager --setopt=appstream*.exclude=*httpd* --save && \
+      dnf -y --disableplugin=subscription-manager module enable mod_auth_openidc && \
+      dnf -y --disableplugin=subscription-manager install mod_auth_openidc; \
+    else \
+      dnf -y install \
+        /opt/app-root/src/bin-rpm-dir/cjose-0.6*.s390x.rpm \
```
Comment (bdunne) [inline Dockerfile:28]: Do we need to add the version numbers here?

Comment (Fryguy) [reply Dockerfile:28]: This was sort of discussed in the related postgres PR: https://github.com/ManageIQ/container-postgresql/pull/39#discussion_r837566634

**Review by bdunne — APPROVED** [review]

**Pull request merged**